core/templates/dev/head/pages/exploration_player/SupplementalCardDirective.js

```diff
@@ -143,6 +155,12 @@ oppia.directive('supplementalCard', [
           };
 
           $scope.$on(EVENT_ACTIVE_CARD_CHANGED, function() {
```

Comment:
Hm. When does this get fired? Does it get fired when navigating back through the progress dots at the top of the page?

Reply:
Yep, it does. But looks like the `HintManagerService.getNumHintsConsumed() > 0` condition isn't satisfied when navigating back using the progress dots. So this isn't a problem I guess.

Reply:
It is a problem for maintainability. The event should only be fired when a new card is added at the end.

Reply:
Can I add a note? Not sure how else to detect progression to a new card.

Reply:
Or I could just call this inside updateActiveCard

Reply:
Nope, this needs to be done correctly. There must be somewhere that handles adding new cards to the end.